### PR TITLE
Layout background color

### PR DESCRIPTION
**Pull request by yogevbd**
* Modified `layout.backgroundColor` to control the layout parent background color.
* Add `layout.componentBackgroundColor` to control the component background color. (Until now on iOS, `layout.backgroundColor` controlled the component and not the parent.
* Remove drawBehind deprecation

### CODE REVIEW

File: lib/ios/RNNBottomTabsPresenter.m [hunk 1]
```diff
@@ -25,6 +25,7 @@ - (void)applyOptions:(RNNNavigationOptions *)options {
     [bottomTabs setTabBarHideShadow:[withDefault.bottomTabs.hideShadow getWithDefaultValue:NO]];
     [bottomTabs setTabBarStyle:[RCTConvert UIBarStyle:[withDefault.bottomTabs.barStyle getWithDefaultValue:@"default"]]];
     [bottomTabs setTabBarVisible:[withDefault.bottomTabs.visible getWithDefaultValue:YES] animated:[withDefault.bottomTabs.animate getWithDefaultValue:NO]];
+    [bottomTabs.view setBackgroundColor:withDefault.layout.backgroundColor.get];
```
Comment (guyca) [inline lib/ios/RNNBottomTabsPresenter.m]: Shouldn't we apply background color with a default value?

Comment (guyca) [reply lib/ios/RNNBottomTabsPresenter.m]: getWithDefaultValue:nil like in SideMenu

File: lib/ios/RNNComponentPresenter.m [hunk 2]
```diff
@@ -127,15 +127,15 @@ - (void)mergeOptions:(RNNNavigationOptions *)options resolvedOptions:(RNNNavigat
 	if (options.topBar.largeTitle.visible.hasValue) {
 		[viewController setTopBarPrefersLargeTitle:options.topBar.largeTitle.visible.get];
 	}
-
+    
+    if (options.layout.componentBackgroundColor.hasValue) {
+        [viewController setBackgroundColor:options.layout.componentBackgroundColor.get];
```
Comment (pie6k) [inline lib/ios/RNNComponentPresenter.m]: Took me ~1hour after upgrading why my overlays arent transparent. It was espetially missleading as typescript file says

```
/**
     * Set background color only for components, helps reduce overdraw if background color is set in default options.
     * #### (Android specific)
     */
    componentBackgroundColor?: Color;
```

So I was ignoring it. Does it mean `backgroundColor` is now useless in component layout?

Comment (andrejcesen) [reply lib/ios/RNNComponentPresenter.m:132]: Hey @pie6k, you're pretty much on point. As long as you use component layout on its own (without being wrapped in stack/bottomTabs/...), only componentBackgroundColor takes effect. But if you use `stack` for example, `backgroundColor` still changes the root UINavigationController background color, so this affects things like topBar (if you set its color to 'transparent', then `backgroundColor` will show through).
Hopefully this clears things up :)

**Pull request merged**